packages/components/src/presentation-components/properties/UniquePropertyValuesSelector.tsx

```diff
@@ -0,0 +1,197 @@
+      return undefined;
+    }
+    const uniqueValues: DisplayValueGroup[] = [];
+    for (let i = 0; i < deserializedDisplayValues.length; i++) {
+      uniqueValues.push({ displayValue: deserializedDisplayValues[i], groupedRawValues: deserializedGroupedRawValues[i] });
+    }
+    return uniqueValues;
+  }
+  return undefined;
+}
+
+interface UseUniquePropertyValuesRulesetProps {
+  descriptor: Descriptor;
+  field?: Field;
+  imodel: IModelConnection;
+}
+
+function useUniquePropertyValuesRuleset({ descriptor, field, imodel }: UseUniquePropertyValuesRulesetProps) {
+  const [ruleset, setRuleset] = useState<Ruleset>();
+  useEffect(() => {
+    void (async () => {
+      const [schemaName, className] = await getSchemaAndClassNames({ imodel, descriptor, field });
+      setRuleset({
+        id: "unique-property-values",
+        rules: [
+          {
+            ruleType: RuleTypes.Content,
+            specifications: [
+              {
+                specType: ContentSpecificationTypes.ContentInstancesOfSpecificClasses,
+                classes: { schemaName, classNames: [className], arePolymorphic: true },
+              },
+            ],
+          },
+        ],
+      });
+    })();
+  }, [field, descriptor, imodel]);
+
+  return ruleset;
+}
+
+async function getSchemaAndClassNames({ imodel, descriptor, field }: UseUniquePropertyValuesRulesetProps) {
+  if (field?.parent === undefined && field?.isPropertiesField()) {
+    return field.properties[0].property.classInfo.name.split(":");
+  }
+  return (
+    await findBaseExpressionClass(
+      imodel,
+      descriptor.selectClasses.map((item) => item.selectClassInfo),
+    )
+  ).name.split(":");
+}
+
+interface UseUniquePropertyValuesLoaderProps {
+  imodel: IModelConnection;
+  ruleset?: Ruleset;
+  fieldDescriptor?: FieldDescriptor;
+}
+
+function useUniquePropertyValuesLoader({ imodel, ruleset, fieldDescriptor }: UseUniquePropertyValuesLoaderProps) {
+  const loadTargets = useCallback(
+    async (loadedOptionsCount: number) => {
+      if (!ruleset || !fieldDescriptor) {
+        return { options: [], hasMore: false };
+      }
+
+      const content = await Presentation.presentation.getPagedDistinctValues({
+        imodel,
+        descriptor: {},
+        fieldDescriptor,
+        rulesetOrId: ruleset,
+        paging: { start: loadedOptionsCount, size: UNIQUE_PROPERTY_VALUES_BATCH_SIZE },
+        keys: new KeySet(),
+      });
+      return {
+        options: content.items.filter(
+          (item) =>
+            item.groupedRawValues.filter((value) => value !== undefined && value !== null) && item.displayValue !== undefined && item.displayValue !== "",
+        ),
+        hasMore: content.items.length === UNIQUE_PROPERTY_VALUES_BATCH_SIZE,
+      };
+    },
+    [imodel, ruleset, fieldDescriptor],
+  );
+
+  return loadTargets;
+}
+
+function serializeDisplayValueGroupArray(values: DisplayValueGroup[]) {
```

Comment:
I would suggest moving this to `common/Utils.ts` so both `serialize` and `deserialize` function are at the same file.

Reply:
fixed